lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -487,6 +481,97 @@ namespace {
   };
 }
 
+static bool fixMissingParameters(InFlightDiagnostic &diag, ValueDecl *decl,
+                                 ValueDecl *baseDecl) {
+  // Only check class override
+  if (!(decl->getDeclContext()->getSelfClassDecl() &&
+      baseDecl->getDeclContext()->getSelfClassDecl())) {
+    return false;
+  }
+  
+  const ParameterList *derivedParams = nullptr;
+  const ParameterList *baseParams = nullptr;
+  if ((isa<AbstractFunctionDecl>(decl) &&
+       isa<AbstractFunctionDecl>(baseDecl)) ||
+      isa<SubscriptDecl>(baseDecl)) {
+    derivedParams = getParameterList(const_cast<ValueDecl *>(decl));
+    baseParams = getParameterList(const_cast<ValueDecl *>(baseDecl));
+  }
+
+  if (!derivedParams && !baseParams) {
+    return false;
+  }
+
+  int diff = baseParams->size() - derivedParams->size();
+  if (diff <= 0)
+    return false;
+  
+  unsigned numMissing = abs(diff);
+  
+  SmallString<32> scratch;
+  llvm::raw_svector_ostream fixIt(scratch);
+
+  auto subs = SubstitutionMap::getOverrideSubstitutions(baseDecl, decl,
+                                                        /*derivedSubs=*/None);
+  
+  bool isFirst = true;
+  for (unsigned i = baseParams->size() - numMissing, n = baseParams->size();
+       i != n; ++i) {
+    auto *baseParam = baseParams->get(i);
+    auto baseParamTy = baseParam->getInterfaceType();
+    if (baseParam->isInOut())
+      baseParamTy = baseParamTy->getInOutObjectType();
+    baseParamTy = baseParamTy.subst(subs);
+      
+    auto argName = baseParam->getArgumentName();
+    auto paraName = baseParam->getParameterName();
+    
+    if (isFirst) {
+      isFirst = false;
+      fixIt << ", ";
+    }
+    
+    if (paraName == argName) {
```

Comment:
Wouldn't something like 
```cpp
auto name = (paraName != argName) ? paraName : "";
// e.g. func foo(a _: Int) {}
// this doesn't make sense, maybe should be improve.
if (argName.empty()) {
 fixIt << "_ " << paraName << ": ";
} else {
 fixIt << argName << " " << paraName << ": ";
}
```
Resolve this case in a simpler way? Just a suggestion ... not sure it would be the case :)